de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/process/BPartnerCreditLimit_AddRemoveCreditStopStatus.java

```diff
@@ -0,0 +1,68 @@
+package org.adempiere.bpartner.process;
+
+import org.adempiere.bpartner.service.BPartnerStats;
+import org.adempiere.bpartner.service.IBPartnerStatsBL;
+import org.adempiere.bpartner.service.IBPartnerStatsBL.CalculateSOCreditStatusRequest;
+import org.adempiere.bpartner.service.IBPartnerStatsDAO;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.Services;
+import org.adempiere.util.time.SystemTime;
+import org.compiere.model.I_C_BPartner_Stats;
+import org.compiere.model.X_C_BPartner_Stats;
+
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.IProcessPreconditionsContext;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+
+public class BPartnerCreditLimit_AddRemoveCreditStopStatus extends JavaProcess implements IProcessPrecondition
+{
+	final IBPartnerStatsBL bpartnerStatsBL = Services.get(IBPartnerStatsBL.class);
+	final IBPartnerStatsDAO bpartnerStatsDAO = Services.get(IBPartnerStatsDAO.class);
+
+	@Param(parameterName = "IsSetCreditStop", mandatory = true)
+	private boolean isSetCreditStop;
+
+	@Override
+	protected String doIt()
+	{
+		final I_C_BPartner_Stats bpStats = getRecord(I_C_BPartner_Stats.class);
+		final String creditStatus;
+		if (isSetCreditStop)
+		{
+			creditStatus = X_C_BPartner_Stats.SOCREDITSTATUS_CreditStop;
+		}
+		else
+		{
+			final BPartnerStats stats = bpartnerStatsDAO.getCreateBPartnerStats(bpStats.getC_BPartner());
+			final CalculateSOCreditStatusRequest request = CalculateSOCreditStatusRequest.builder()
```

Comment:
wht does this code do? pls outline it briefly in the process javadoc and/or here and/or at CalculateSOCreditStatusRequest and/or at the calculateSOCreditStatus() javadoc